executor/set_test.go

```diff
@@ -529,10 +529,10 @@ func (s *testSerialSuite1) TestSetVar(c *C) {
 	tk.MustExec("set @@tidb_allow_fallback_to_tikv = 'tiflash, tiflash, tiflash'")
 	tk.MustQuery("select @@tidb_allow_fallback_to_tikv").Check(testkit.Rows("tiflash"))
 
-	tk.MustGetErrMsg("SET SESSION tidb_allow_fallback_to_tikv = 'tikv,tiflash'", "ERROR 1231 (42000): Variable 'tidb_allow_fallback_to_tikv' can't be set to the value of 'tikv,tiflash'")
-	tk.MustGetErrMsg("SET GLOBAL tidb_allow_fallback_to_tikv = 'tikv,tiflash'", "ERROR 1231 (42000): Variable 'tidb_allow_fallback_to_tikv' can't be set to the value of 'tikv,tiflash'")
-	tk.MustGetErrMsg("set @@tidb_allow_fallback_to_tikv = 'tidb, tiflash, tiflash'", "ERROR 1231 (42000): Variable 'tidb_allow_fallback_to_tikv' can't be set to the value of 'tidb, tiflash, tiflash'")
-	tk.MustGetErrMsg("set @@tidb_allow_fallback_to_tikv = 'unknown, tiflash, tiflash'", "ERROR 1231 (42000): Variable 'tidb_allow_fallback_to_tikv' can't be set to the value of 'unknown, tiflash, tiflash'")
+	tk.MustGetErrMsg("SET SESSION tidb_allow_fallback_to_tikv = 'tikv,tiflash'", "[variable:1231]Variable 'tidb_allow_fallback_to_tikv' can't be set to the value of 'tikv,tiflash'")
```

Comment:
The error message should also point out the permitted value.

Reply:
The error message is for all wrongly setting some system variable, which is consistent with MySQL(https://github.com/pingcap/tidb/blob/master/errno/errname.go#L243). If we want to change the error message for wrongly setting `tidb_allow_fallback_to_tikv`, we might need to choose another error code.